app/models/book.rb

```diff
@@ -0,0 +1,61 @@
+class Book < ApplicationRecord
+  CATEGORIES = %w[mobile_development photo web_design web_development].freeze
+
+  has_and_belongs_to_many :authors
+  has_and_belongs_to_many :categories
+  has_many :images, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+  has_many :line_items
+
+  before_destroy :ensure_not_referenced_by_any_line_item
+
+  validates :title, :description, :price, :publication_year,
+            :dimensions, :materials, presence: true
+  validates :price, numericality: { greater_than_or_equal_to: 0.01 }
+  validates :title, uniqueness: { case_sensitive: false }
+  validates :publication_year, inclusion: { in: 1969..Date.today.year }
+
+  paginates_per 12
+
+  scope :mobile_development, -> do
+    Book.joins(:categories).where('categories.name = ?', 'Mobile development')
+  end
+
+  scope :photo, -> do
+    Book.joins(:categories).where('categories.name = ?', 'Photo')
+  end
+
+  scope :web_design, -> do
+    Book.joins(:categories).where('categories.name = ?', 'Web design')
+  end
+
+  scope :web_development, -> do
+    Book.joins(:categories).where('categories.name = ?', 'Web development')
+  end
+
+  scope :best_seller, ->(category) do
+    return Book.none unless CATEGORIES.include?(category.to_s)
+    return Book.public_send(category).first unless LineItem.any?
```

Comment:
@b-artem the same comment as about any? above

Reply:
Done